packages/ra-core/src/controller/create/CreateBase.tsx

```diff
@@ -40,28 +41,45 @@ import { ResourceContextProvider } from '../../core';
 export const CreateBase = <
     RecordType extends Omit<RaRecord, 'id'> = any,
     ResultRecordType extends RaRecord = RecordType & { id: Identifier },
+    MutationOptionsError = Error,
 >({
     children,
+    loading = null,
     ...props
-}: CreateControllerProps<RecordType, Error, ResultRecordType> & {
-    children: ReactNode;
-}) => {
+}: CreateBaseProps<RecordType, ResultRecordType, MutationOptionsError>) => {
     const controllerProps = useCreateController<
         RecordType,
-        Error,
+        MutationOptionsError,
         ResultRecordType
     >(props);
-    const body = (
-        <CreateContextProvider value={controllerProps}>
-            {children}
-        </CreateContextProvider>
-    );
-    return props.resource ? (
-        // support resource override via props
-        <ResourceContextProvider value={props.resource}>
-            {body}
-        </ResourceContextProvider>
-    ) : (
-        body
+
+    const isAuthPending = useIsAuthPending({
+        resource: controllerProps.resource,
+        action: 'create',
+    });
+
+    if (isAuthPending && !props.disableAuthentication) {
+        return loading;
+    }
+
+    return (
+        <OptionalResourceContextProvider value={controllerProps.resource}>
```

Comment:
```suggestion
        <OptionalResourceContextProvider value={props.resource}>
```

Otherwise it's a BC. Same for other controllers.

Reply:
No because the controller already handle it